cmd/proxy/main.go

```diff
@@ -38,15 +41,37 @@ func main() {
 		log.Fatal(err)
 	}
 
-	cert, key, err := conf.TLSCertFiles()
+	cert, key, mtlsCert, err := conf.TLSCertFiles()
 	if err != nil {
 		log.Fatal(err)
 	}
+	var srv *http.Server
 
-	srv := &http.Server{
-		Addr:    conf.Port,
-		Handler: handler,
+	if mtlsCert != "" {
+		clientCACert, err := ioutil.ReadFile(conf.TLSCACertFile)
```

Comment:
`ioutil` was depreciated in Go 1.16 and is not safe to use. Use `os.ReadFile` instead.